articles/prototyping-machine-learning-models-with-streamlit/index.md

```diff
@@ -0,0 +1,212 @@
+ 
+##### Streamlit Set up
+ 
+The installation is very simple, just run this on your command prompt or terminal
+ 
+```python
+pip install streamlit
+streamlit hello
+```
+ 
+To import it, use the simple
+ 
+```python
+import pandas as pd
+import streamlit as st
+import plotly.express as px
+```
+ 
+And to run your streamlit app use
+ 
+```python
+streamlit run app.py
+```
+ 
+#### Building Sentiment Analyzer
+ 
+Now for this beginner’s tutorial, we will be building a sentiment analyzer model and prototype it with streamlit. The first step will be getting a Dataset, and the dataset can be found [HERE](https://drive.google.com/drive/folders/1wk0hkrRhxe6t5g390g5V5O-CUcRB6BZC?usp=sharing), next let’s create a directory and create a virtual environment, open you terminal or command prompt and type in the code below.
+ 
+```python
+$ mkdir textanalyzer
+$ cd textanalyzer
+```
+ 
+Nowe for us to create the virtual environment we use the code below this to isolate the environment.
+ 
+```python
+$ python3.8 -m venv env
+$ source env/bin/activate
+```
+ 
+The first line of code creates the virtual environment while the second activate it. Now copy the dataset as it is in the folder and paste it into this textanalyzer directory, and then create a file and save it as sentiment_analyzer.py, this will be where we will be writing our code to build the streamlit app. The next thing will be to import all the packages we will be using for this tutorial.
+ 
+```python
+import streamlit as st
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.naive_bayes import BernoulliNB
+from sklearn.metrics import confusion_matrix
+import numpy as np
+import itertools
+import matplotlib.pyplot as plt
+```
+ 
+After that let’s add a header for our web app and a subheader and to do that we use the code below
+ 
+```python
+st.title("Sentiment Analyzer Based On Text Analysis ")
+st.subheader("Paras Patidar - MLAIT")
+st.write('\n\n')
+```
+ 
+The next step will be to write a function that will get all the data from our directory and bring them up, your directory should look like
+ 
+![Simple](https://github.com/jamessandy/engineering-education/blob/new-article/articles/prototyping-machine-learning-models-with-streamlit/img1.jpg)
+ 
+```python
+@st.cache
+def get_all_data():
+ root = "Datasets/"
+ with open(root + "imdb_labelled.txt", "r") as text_file:
+     data = text_file.read().split('\n')
+ with open(root + "amazon_cells_labelled.txt", "r") as text_file:
+     data += text_file.read().split('\n')
+ with open(root + "yelp_labelled.txt", "r") as text_file:
+     data += text_file.read().split('\n')
+ return data
+```
+ 
+If you notice in the function above the first line is *@st.cache*, this mark tells streamlit that whenever the function is called that there are some certain requirements it should check for and they should be fulfilled, now create a variable called all_data which should call the *get_all_data* function we wrote.
```

Comment:
Notice the code snippet @st.cache in the first line. This tells streamlit that whenever the function is called, some requirements should be fulfilled. Create a new variable called `all_data.` `all_data` calls the function `get_all_data` we defined earlier.